pkg/apis/playlist/v0alpha1/legacy_storage.go

```diff
@@ -109,3 +112,88 @@ func (s *legacyStorage) Get(ctx context.Context, name string, options *metav1.Ge
 
 	return convertToK8sResource(dto, s.namespacer), nil
 }
+
+func (s *legacyStorage) Create(ctx context.Context,
+	obj runtime.Object,
+	createValidation rest.ValidateObjectFunc,
+	options *metav1.CreateOptions,
+) (runtime.Object, error) {
+	info, err := grafanarequest.NamespaceInfoFrom(ctx, true)
+	if err != nil {
+		return nil, err
+	}
+
+	p, ok := obj.(*Playlist)
+	if !ok {
+		return nil, fmt.Errorf("expected playlist?")
+	}
+	cmd, err := convertToLegacyUpdateCommand(p, info.OrgID)
+	if err != nil {
+		return nil, err
+	}
+	out, err := s.service.Create(ctx, &playlist.CreatePlaylistCommand{
+		UID:      p.Name,
+		Name:     cmd.Name,
+		Interval: cmd.Interval,
+		Items:    cmd.Items,
+		OrgId:    cmd.OrgId,
+	})
+	if err != nil {
+		return nil, err
+	}
+	return s.Get(ctx, out.UID, nil)
+}
+
+func (s *legacyStorage) Update(ctx context.Context,
+	name string,
+	objInfo rest.UpdatedObjectInfo,
+	createValidation rest.ValidateObjectFunc,
+	updateValidation rest.ValidateObjectUpdateFunc,
+	forceAllowCreate bool,
+	options *metav1.UpdateOptions,
+) (runtime.Object, bool, error) {
+	info, err := grafanarequest.NamespaceInfoFrom(ctx, true)
+	if err != nil {
+		return nil, false, err
+	}
+
+	created := false
+	old, err := s.Get(ctx, name, nil)
+	if err != nil {
+		return old, created, err
+	}
+
+	obj, err := objInfo.UpdatedObject(ctx, old)
+	if err != nil {
+		return old, created, err
+	}
+	p, ok := obj.(*Playlist)
+	if !ok {
+		return nil, created, fmt.Errorf("expected playlist after update")
+	}
+
+	cmd, err := convertToLegacyUpdateCommand(p, info.OrgID)
+	if err != nil {
+		return old, created, err
+	}
+	_, err = s.service.Update(ctx, cmd)
+	if err != nil {
+		return nil, false, err
+	}
+
+	r, err := s.Get(ctx, name, nil)
+	return r, created, err
+}
+
+// GracefulDeleter
+func (s *legacyStorage) Delete(ctx context.Context, name string, deleteValidation rest.ValidateObjectFunc, options *metav1.DeleteOptions) (runtime.Object, bool, error) {
```

Comment:
delete seems to have the same response if the object is actually deleted or not.

```
kubectl get playlists
NAME                                   CREATED AT
afca8893-31df-4c13-9d8e-bde0f1591ca5   2023-10-23T21:39:31Z
```

```
kubectl delete playlists afca8893-31df-4c13-9d8e-bde0f1591ca5
playlist.playlist.x.grafana.com "afca8893-31df-4c13-9d8e-bde0f1591ca5" deleted
Error from server: Playlist not found
```

```
kubectl delete playlists does-not-exist  
playlist.playlist.x.grafana.com "does-not-exist" deleted
Error from server: Playlist not found
```

Reply:
Interesting!  looks like there are internal calls to "get" after delete, and the error must support `errors.IsNotFound()` otherwise things are funky